### PR TITLE
execution: fix scalar function

### PR DESCRIPTION
**Pull request by MichaHoffmann**
this should fix issues with `scalar(non_existent)` problems in fuzzing.

### CODE REVIEW

File: execution/function/scalar.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		return nil, nil
+	}
+
+	in, err := o.next.Next(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	result := o.GetPool().GetVectorBatch()
+	if len(in) == 0 {
```
Comment (fpetkovski) [inline execution/function/scalar.go:53]: We should not need this anymore right?

Comment (MichaHoffmann) [reply execution/function/scalar.go:53]: ok rebased, it works as intended :tada:

File: execution/function/scalar.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	if err != nil {
+		return nil, err
+	}
+	if len(in) == 0 {
+		return nil, nil
+	}
+
+	result := o.GetPool().GetVectorBatch()
+	for _, vector := range in {
+		sv := o.GetPool().GetStepVector(vector.T)
```
Comment (fpetkovski) [inline execution/function/scalar.go:53]: I noticed that in the original implementation we modify the input vector in place: https://github.com/thanos-io/promql-engine/pull/265/files#diff-45b283ce3f8010b8cd1c8c50b568e54e58a974d8b4ba75b14e8821680641b0fcL219.

Can we do the same here?

Comment (MichaHoffmann) [reply execution/function/scalar.go:53]: i think we could do that yes

File: execution/function/scalar.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	default:
+	}
+	in, err := o.next.Next(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if len(in) == 0 {
```
Comment (fpetkovski) [inline execution/function/scalar.go:47]:
```suggestion
	if in == nil {
```

**Review by fpetkovski — APPROVED** [review]
Looks great 👍

File: execution/function/scalar.go (hunk 2 above)
Comment (MichaHoffmann) [reply execution/function/scalar.go:48]: note: tried and it did fail some range query tests; decided to go back to using the pool from the scalar operator itself.

**Pull request merged**